pkg/controller/kfservice/reconcilers/knative/service_reconciler.go

```diff
@@ -167,7 +167,6 @@ func (r *ServiceReconciler) reconcileService(kfsvc *v1alpha2.KFService, desired
 	log.Info("Updating service", "namespace", desired.Namespace, "name", desired.Name)
 	existing.Spec = desired.Spec
 	existing.ObjectMeta.Labels = desired.ObjectMeta.Labels
-	existing.ObjectMeta.Annotations = desired.ObjectMeta.Annotations
```

Comment:
We still need to pass down the annotation from kfservice though, maybe we can check if annotation from desired is in existing annotations we overwrite otherwise append.

Reply:
For now, desired.ObjectMeta.Annotations is alway empty
maybe we can pass down the annotation when we really set something in desired.ObjectMeta.Annotations in future

Reply:
We do set a bunch of annotations here https://github.com/kubeflow/kfserving/blob/master/pkg/controller/kfservice/resources/knative/service.go#L107.

Reply:
@hougangliu you are right, the annotation is set on revision template not on the knative service itself and this was originally added by you :) (https://github.com/kubeflow/kfserving/pull/64/files). I think I am ok to take off this considering this is conflicting with knative internal annotations.